src/library/dlgtagfetcher.cpp

```diff
@@ -246,49 +306,105 @@ void DlgTagFetcher::fetchTagFinished(
-        int httpError, const QString& app, const QString& message, int code) {
+        int httpError,
+        const QString& app,
+        const QString& message,
+        int code) {
     m_networkResult = httpError == 0 ? NetworkResult::UnknownError : NetworkResult::HttpError;
     m_data.m_pending = false;
-    QString strError = tr("HTTP Status: %1");
-    QString strCode = tr("Code: %1");
-    httpStatus->setText(strError.arg(httpError) + "\n" + strCode.arg(code) + "\n" + message);
-    QString unknownError = tr("Mixxx can't connect to %1 for an unknown reason.");
-    cantConnectMessage->setText(unknownError.arg(app));
-    QString cantConnect = tr("Mixxx can't connect to %1.");
-    cantConnectHttp->setText(cantConnect.arg(app));
-    updateStack();
+
+    if (m_networkResult == NetworkResult::UnknownError) {
+        QString unknownError = tr("Can't connect to %1 for an unknown reason: %2");
+        loadingProgressBar->setFormat(unknownError.arg(app, message));
+    } else {
+        QString cantConnect = tr("Can't connect to %1: %2");
+        loadingProgressBar->setFormat(cantConnect.arg(app, message));
+    }
+    qWarning() << "Error while getting track metadata!"
+               << "Service:" << app
+               << "HTTP Status:" << httpError
+               << "Code:" << code
+               << "Server message:" << message;
+    btnRetry->setEnabled(true);
+    updateView();
+}
+
+void DlgTagFetcher::updateOriginalTag() {
+    results->clear();
+
+    addDivider(tr("Original tags"), results);
+    addTrack(trackColumnValues(*m_track), kOriginalTrackIndex, results);
+
+    addDivider(tr("Suggested tags"), results);
+    {
+        int trackIndex = 0;
+        QSet<QStringList> allColumnValues; // deduplication
+        for (const auto& trackRelease : qAsConst(m_data.m_results)) {
+            const auto columnValues = trackReleaseColumnValues(trackRelease);
+            // Ignore duplicate results
+            if (!allColumnValues.contains(columnValues)) {
+                allColumnValues.insert(columnValues);
+                addTrack(columnValues, trackIndex, results);
+            }
+            ++trackIndex;
+        }
+    }
+
+    for (int i = 0; i < results->model()->rowCount(); ++i) {
+        const QModelIndex index = results->model()->index(i, 0);
+        const QVariant id = index.data(Qt::UserRole);
+        if (!id.isNull() && id.toInt() == m_data.m_selectedResult) {
+            results->setCurrentIndex(index);
+            break;
+        }
+    }
+}
+
+void DlgTagFetcher::createView() {
+    results->clear();
+    addDivider(tr("Original tags"), results);
+    addTrack(trackColumnValues(*m_track), kOriginalTrackIndex, results);
+    btnApply->setDisabled(true);
+    successMessage->setVisible(false);
+    loadingProgressBar->setVisible(true);
+    loadingProgressBar->setValue(kPercentOfConstantTask);
 }
 
-void DlgTagFetcher::updateStack() {
+void DlgTagFetcher::updateView() {
```

Comment:
This still sound like function that can be called unconditionally repeatedly. But that is not true, because it must be only called after a ConstantTask for my understanding. But I see that this is called from three places. Is that what your are testing in the if branches below? 
 
progressBarSetCurrentStep() Is also confusing, because it is not universal for all "Steps" only for recordings. Is that correct?

Reply:
Can we remove the entire updateView() and copy its content to the callers?

Reply:
Done 👍